### PR TITLE
tools should have its own Maven module

### PR DESCRIPTION
**Pull request by vladak**
For now this is only for the tests to run separately. In the future this could be used to properly deliver tools and have them consumed properly by the `distribution` module.

### CODE REVIEW

**Comment by Orviss** [comment]
Just a suggestion: could we move `.py` files to `src/main/python` directory? I found that Apache Spark does it similarly and it should make the structure more consistent (<https://github.com/apache/spark/tree/master/examples/src/main/python>). 

However, I have never used Maven with other languages so maybe it should be done differently.

**Comment by vladak** [comment]
What advantage this will have w.r.t. Maven ?

**Comment by Orviss** [comment]
> What advantage this will have w.r.t. Maven ?

I'm not sure. I don't know if you need some Maven plugins or something else to make it work properly.

However, for me, it would look nice that the source code would have the same directory structure. Imagine that tools module were to contain some Java code now. Then it would be separated nicely into `src/main/java` and `src/main/python`.

**Comment by vladak** [comment]
Makes sense, also for the tests.

**Comment by vladak** [comment]
Reorganized the structure, had to set the directories in the pom file explicitly. I am not sure how it would work if I added Java/... code there.

**Comment by tulinkry** [comment]
I think eventualy it should be a python package and the distribution version of this package we should ship with each opengrok release.

File: tools/pom.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    <artifactId>tools</artifactId>
+    <version>1.1-rc35</version>
+    <packaging>pom</packaging>
+
+    <name>OpenGrok tools</name>
+
+    <build>
+        <sourceDirectory>src/main/python</sourceDirectory>
+	<testSourceDirectory>src/test/python</testSourceDirectory>
```
Comment (Orviss) [inline tools/pom.xml]: Tab character.

**Pull request merged**